### DISCUSSION TITLE
Enable Gaussion priors

### DISCUSSION THREAD

**Discussion by andreashlarsen** [SasView/sasview#2212] [Ideas]
I suggest to enable Gaussian priors instead of just min and max limits for the fitting parameters (which implicitly assumes uniform priors). We did a proof-of-concept in this paper:
https://doi.org/10.1107/S1600576718008956
(also presented at the SAS conference 2018)

we showed that the method is more stable for noisy data compared to only using uniform priors. We also showed that Gaussian priors prevents overfitting. The error estimates for refined params were also improved - as they reflected the collective knowledge from data and prior knowledge (e.g. from chemical assays prior to the SAXS/SANS experiments). The method is implemented as a web application:
https://somo.chem.utk.edu/bayesfit/
but only with a few models available - those from the paper. So the program is not really useful for the community as such, but at least it shows that the method works. 

I know this would be a lot of work - because it is a rather fundamental change. But I also think it would be the right way to go in SAS model fitting in the long run. 

from a more practical point-of-view: such expansion of sasview would help in those situations, were you set limits for a parameter, and you keep getting either the min or max value for one or more parameters during fitting. It is basically applying restraints instead of constraints, which (in many cases) better reflects and quantifies our prior knowledge.

**Comment by pkienzle** [comment]
Gaussian priors are available if you use sasmodels directly from bumps. [This example](https://github.com/SasView/sasmodels/blob/master/example/model.py) uses box priors, but you could replace `P.range()` with `P.dev(std, mean=mean, limits=limits)` for a truncated Gaussian distribution.

Gaussian priors could be added to the GUI but that will require quite a bit of effort. It will also make the GUI harder to use. Unless many users have good prior information on their fitting parameters this additional complexity may not be worthwhile compared to documentation on how to write the fitting script.
